packages/contracts/test/diamond/facets/DirectGovernanceFarmerFacet.t.sol

```diff
@@ -92,70 +91,180 @@ contract DirectGovernanceFarmerTest is LocalTestHelper {
+        assertTrue(directGovernanceFarmerFacet.isMetaPoolCoin(address(token0)));
+    }
+
+    function testIsMetaPoolCoinReturnTrueIfToken1IsPassed() public {
+        assertTrue(directGovernanceFarmerFacet.isMetaPoolCoin(address(token1)));
+    }
+
+    function testIsMetaPoolCoinReturnTrueIfToken2IsPassed() public {
+        assertTrue(directGovernanceFarmerFacet.isMetaPoolCoin(address(token2)));
+    }
+
+    function testIsMetaPoolCoinReturnTrueIfUbiquityDollarTokenIsPassed()
+        public
+    {
+        assertTrue(directGovernanceFarmerFacet.isMetaPoolCoin(address(dollar)));
+    }
+
+    function testIsMetaPoolCoinReturnFalseIfTokenAddressIsNotInMetaPool()
+        public
+    {
+        assertFalse(directGovernanceFarmerFacet.isMetaPoolCoin(address(0)));
+    }
+
+    function test_Address() public view returns (address) {
+        return address(this);
+    }
+
+    function test_IsMetaPoolCoin() public view {
+        directGovernanceFarmerFacet.isMetaPoolCoin(address(0x2));
+    }
+
     function testDeposit_ShouldRevert_IfTokenIsNotInMetapool() public {
         address userAddress = address(0x100);
         vm.prank(userAddress);
         vm.expectRevert(
             "Invalid token: must be DAI, USD Coin, Tether, or Ubiquity Dollar"
         );
-        directGovernanceFarmer.depositSingle(address(0), 1, 1);
+        directGovernanceFarmerFacet.depositSingle(address(0x250), 1, 1);
     }
 
-    function testDeposit_ShouldRevert_IfAmountIsNotPositive() public {
+    function testDeposit_ShouldRevert_IfDurationIsNotValid() public {
+        address userAddress = address(0x100);
+        vm.prank(userAddress);
+        vm.expectRevert("duration weeks must be between 1 and 208");
+        directGovernanceFarmerFacet.depositSingle(address(token1), 1, 0);
+    }
+
+    function testWithdrawShouldRevertIfTokenIsNotInMetaPool() public {
         address userAddress = address(0x100);
         vm.prank(userAddress);
-        vm.expectRevert("amount must be positive vale");
-        directGovernanceFarmer.depositSingle(address(token0), 0, 1);
+        vm.expectRevert(
+            "Invalid token: must be DAI, USD Coin, Tether, or Ubiquity Dollar"
+        );
+        directGovernanceFarmerFacet.withdraw(1, address(0));
     }
 
-    function testDeposit_ShouldRevert_IfDurationIsNotValid() public {
+    function testDepositMultipleTokens_ShouldRevert_IfAmountsIsNotPositive()
+        public
+    {
+        address userAddress = address(0x100);
+        vm.prank(userAddress);
+        vm.expectRevert("amounts==0");
+        directGovernanceFarmerFacet.depositMulti(
+            [uint256(0), uint256(0), uint256(0), uint256(0)],
+            1
+        );
+    }
+
+    function testDepositMultipleTokens_ShouldRevert_IfDurationIsNotValid()
+        public
+    {
         address userAddress = address(0x100);
         vm.prank(userAddress);
         vm.expectRevert("duration weeks must be between 1 and 208");
```

Comment:
We could also add a boundary value check: 209 weeks

Reply:
so basically changing the revert message

Reply:
but the thing is also the duration can be 1 week only

Reply:
> but the thing is also the duration can be 1 week only

require(
            durationWeeks >= 1 && durationWeeks <= 208,
            "duration weeks must be between 1 and 208"
        );

I meant that in this case we can test not only durationWeeks == 0 but also another boundary value durationWeeks == 209 , perhaps we can add this later in another PR as this one is just moving to facet.